src/foam/u2/wizard/agents/AnalyticEventsAgent.js

```diff
@@ -43,10 +43,10 @@ foam.CLASS({
       var self = this;
       var trace = this.traceIDKey$get(this.__subContext__);
       var obj = this.objectIDKey$get(this.__subContext__);
-      this.analyticsAgent.sub('event', function(_, __, ___, evt) {
+      this.analyticsAgent.sub('event', async function(_, __, ___, evt) {
         // TODO: add subclass support
         let analyticEvent = self.AnalyticEvent.create({...evt, traceId: trace, objectId: obj, sessionId: self.sessionID})
-        self.analyticEventDAO.put(analyticEvent);
+        await self.analyticEventDAO.put(analyticEvent);
```

Comment:
Is this the only put that we need to ensure completes before sending final status (SUCCESS / CANCELLED)?

Reply:
What does the async and await change?